sycl/include/CL/sycl/handler.hpp

```diff
@@ -522,6 +522,79 @@ class __SYCL_EXPORT handler {
     return true;
   }
 
+#ifndef __SYCL_DEVICE_ONLY__
+  /// Copies the content of memory object accessed by Src into the memory
+  /// pointed by Dst.
+  ///
+  /// \param Src is a source SYCL accessor.
+  /// \param Dst is a pointer to destination memory.
+  template <typename TSrc, typename TDst, int Dim, access::mode AccMode,
+            access::target AccTarget, access::placeholder IsPH>
+  detail::enable_if_t<(Dim > 0)>
+  copyAccToPtrHost(accessor<TSrc, Dim, AccMode, AccTarget, IsPH> Src,
+                   TDst *Dst) {
+    range<Dim> Range = Src.get_range();
+    parallel_for<class __copyAcc2Ptr<TSrc, TDst, Dim, AccMode, AccTarget, IsPH>>
+        (Range, [=](id<Dim> Index) {
```

Comment:
Can't we merge two functions(lines 534 and 554) into one by using "constexpr int AdjustedDim = Dim == 0 ? 1 : Dim" instead of just Dim on lines 536 and 538?

Reply:
I don't see how that can be done.
The 0-dimensional accessor does not have 'get_range()' method used on line 536. Also, 0-dim accessor does not support 'operator[]' used on the line 542

Reply:
Ok.